internal/export/exportfile_test.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package export
+
+import (
+	"crypto"
+	"io"
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/google/exposure-notifications-server/internal/export/model"
+	"github.com/google/exposure-notifications-server/internal/pb/export"
+	publishmodel "github.com/google/exposure-notifications-server/internal/publish/model"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+)
+
+func TestMarshalUnmarshalExportFile(t *testing.T) {
+	tmp, err := ioutil.TempDir("", "")
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Cleanup(func() { os.RemoveAll(tmp) })
+
+	batchStartTime := time.Date(2020, 5, 1, 0, 0, 0, 0, time.UTC).Truncate(time.Second)
+	batchEndTime := batchStartTime.Add(1 * time.Hour)
+
+	batch := &model.ExportBatch{
+		BatchID:          1,
+		ConfigID:         1,
+		BucketName:       "test-bucket",
+		FilenameRoot:     "files",
+		StartTimestamp:   batchStartTime,
+		EndTimestamp:     batchEndTime,
+		Region:           "US",
+		Status:           "",
+		LeaseExpires:     time.Time{},
+		SignatureInfoIDs: []int64{1, 2},
+	}
+
+	exposures := []*publishmodel.Exposure{
+		{
+			ExposureKey:      []byte("ABC"),
+			Regions:          []string{"US"},
+			IntervalNumber:   18,
+			IntervalCount:    0,
+			CreatedAt:        batchStartTime,
+			LocalProvenance:  true,
+			TransmissionRisk: 8,
+		},
+		{
+			ExposureKey:      []byte("DEF"),
+			Regions:          []string{"CA"},
+			IntervalNumber:   118,
+			IntervalCount:    1,
+			CreatedAt:        batchEndTime,
+			LocalProvenance:  true,
+			TransmissionRisk: 1,
+		},
+	}
+
+	signatureInfo := &model.SignatureInfo{
+		SigningKey:        "/kms/project/key/1",
+		SigningKeyVersion: "1",
+		SigningKeyID:      "310",
+		EndTimestamp:      batchEndTime,
+	}
+
+	signer := &customTestSigner{}
+
+	blob, err := MarshalExportFile(batch, exposures, 1, 1, []ExportSigners{
+		{SignatureInfo: signatureInfo,
+			Signer: signer}})
+	if err != nil {
+		t.Fatalf("Can't marshal export file, %v", err)
+	}
+
+	filePath := tmp + "/test-export.zip"
+	err = ioutil.WriteFile(filePath, blob, 0644)
```

Comment:
I suppose the test doesn't need actually to exercise writing to storage/reading from storage to verify marshalling/unmarshalling. That might make it more flexible where it runs, but I'm not fully up to speed on our continuous integration testing story so this might be moot.